lib/spack/spack/variant.py

```diff
@@ -510,6 +491,18 @@ def satisfies(self, other, strict=False):
         return all(k in self and self[k].satisfies(other[k])
                    for k in to_be_checked)
 
+    def compatible(self, other):
+        """Returns True iff it's possible to constrain this VariantMap by other
+
+        Args:
+            other (VariantMap): VariantMap instance to check against
+
+        Returns:
+            bool
+        """
+        to_be_checked = filter(lambda x: x in self, other.keys())
```

Comment:
Could you use `&` for sets here like: `to_be_checked = set(self.keys()) & set(other.keys())`